server.go

```diff
@@ -456,6 +456,15 @@ func (srv *Server) serveTCPconnection(ctx *shutdownContext, netConn net.Conn) er
 
 		msg.fill(mti, o, p)
 
+		size, err := msg.ToBytesLength()
+		if err != nil {
+			return session.closeWithError(fmt.Errorf("cannot serve tcp connection: %v", err))
+		}
+
+		if uint32(size) > srv.MaxMessageSize {
```

Comment:
pls move this condition to line 440(before `body := make([]byte, mti.BodyLen())` and extends it about `srv.MaxMessageSize != 0 &&` .

Use  `mti.totLen` instead of `msg.ToBytesLength()`